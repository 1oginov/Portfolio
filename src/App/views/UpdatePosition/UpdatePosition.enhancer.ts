import { compose, withHandlers } from 'recompose';

import withPositionById, { Props as WithPositionByIdProps } from 'Portfolio/enhancers/withPositionById';

import withNavigationHandlers from '../../enhancers/withNavigationHandlers';
import withRouteParams from '../../enhancers/withRouteParams';
import * as R from '../../routes';
import { RouteParamsState } from '../../State';
import { Props } from './UpdatePosition';

interface WithNavigationHandlersProps {
  handlePositionClick: (positionId: string) => void;
}

export default compose<Props, {}>(
  withNavigationHandlers({
    handlePositionClick: (props, positionId) => ({
      params: { position: positionId },
      route: R.POSITION,
    }),
  }),
  withRouteParams,
  withPositionById<{ routeParams: RouteParamsState }>(({ routeParams }) => routeParams.position),
  withHandlers<WithNavigationHandlersProps & WithPositionByIdProps, {}>({

    handleBackClick: ({ handlePositionClick, position }) => () => {
      if (position) {
        handlePositionClick(position.id);
      }
    },

  }),
);
