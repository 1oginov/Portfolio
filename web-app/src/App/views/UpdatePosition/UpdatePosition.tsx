import { Typography } from '@material-ui/core';
import * as React from 'react';

import { Position } from 'Layers/Business/Services/PortfolioService/Interfaces/Position';
import { Progress } from 'Layers/Presentation/Components/Progress';
import { UpdatePositionForm } from 'Layers/Presentation/Components/UpdatePositionForm';

export interface Props {
  classes: { [key: string]: string };
  handleBackClick: () => void;
  position: Position | null;
  positionLoading: boolean;
}

const UpdatePosition: React.FunctionComponent<Props> = ({
  classes, handleBackClick, position, positionLoading,
}: Props) => (
  <div className={classes.root}>

    <Typography className={classes.headline} variant="h4">Update position</Typography>

    {positionLoading || !position
      ? <Progress />
      : (
        <UpdatePositionForm
          onCancelButtonClick={handleBackClick}
          onUpdate={handleBackClick}
          position={position}
        />
      )}

  </div>
);

export default UpdatePosition;
