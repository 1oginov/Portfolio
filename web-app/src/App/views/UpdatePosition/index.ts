import UpdatePosition from './UpdatePosition';
import enhance from './UpdatePosition.enhancer';
import style from './UpdatePosition.styles';

export default style(enhance(UpdatePosition));
