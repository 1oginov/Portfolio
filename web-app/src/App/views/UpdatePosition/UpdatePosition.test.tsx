/* eslint-disable import/no-extraneous-dependencies */

import { shallow } from 'enzyme';
import classes from 'identity-obj-proxy';
import * as React from 'react';

import UpdatePosition from './UpdatePosition';

it('matches snapshot', () => {
  const wrapper = shallow(
    <UpdatePosition
      classes={classes}
      handleBackClick={() => undefined}
      position={{
        amount: 1,
        closeCommission: null,
        closeDate: null,
        closePrice: null,
        id: 'id',
        openCommission: 1,
        openDate: '2019-07-01',
        openPrice: 100,
        symbol: 'AAPL',
      }}
      positionLoading={false}
    />,
  );

  expect(wrapper).toMatchSnapshot();
});

it('matches snapshot when loading', () => {
  const wrapper = shallow(
    <UpdatePosition
      classes={classes}
      handleBackClick={() => undefined}
      position={null}
      positionLoading
    />,
  );

  expect(wrapper).toMatchSnapshot();
});
