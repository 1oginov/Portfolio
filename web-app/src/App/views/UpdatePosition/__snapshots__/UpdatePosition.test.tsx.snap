// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="root"
>
  <WithStyles(ForwardRef(Typography))
    className="headline"
    variant="h4"
  >
    Update position
  </WithStyles(ForwardRef(Typography))>
  <Connect(withHandlers(UpdatePositionForm))
    backButton={
      <WithStyles(ForwardRef(Button))
        color="primary"
        onClick={[Function]}
      >
        Cancel
      </WithStyles(ForwardRef(Button))>
    }
    onClose={[Function]}
    position={
      Object {
        "amount": 1,
        "closeCommission": null,
        "closeDate": null,
        "closePrice": null,
        "id": "id",
        "openCommission": 1,
        "openDate": "2019-07-01",
        "openPrice": 100,
        "symbol": "AAPL",
      }
    }
  />
</div>
`;

exports[`matches snapshot when loading 1`] = `
<div
  className="root"
>
  <WithStyles(ForwardRef(Typography))
    className="headline"
    variant="h4"
  >
    Update position
  </WithStyles(ForwardRef(Typography))>
  <Progress />
</div>
`;
